src/satosa/backends/saml2.py

```diff
@@ -35,6 +35,31 @@
 logger = logging.getLogger(__name__)
 
 
+def get_memorized_idp(context, config, force_authn):
+    memorized_idp = (
+        config.get(SAMLBackend.KEY_MEMORIZE_IDP)
+        and context.state.get(Context.KEY_MEMORIZED_IDP)
+    )
+    use_when_force_authn = config.get(
+        SAMLBackend.KEY_USE_MEMORIZED_IDP_WHEN_FORCE_AUTHN
+    )
+    value = (not force_authn or use_when_force_authn) and memorized_idp
+    return value
+
+
+# XXX check KEY_FORCE_AUTHN value type (boolean vs str)
```

Comment:
We definitely need to do that. The problem here is that this value will be store/serialized in `context.state` when the user will be redirected to the discovery service. Since JSON has no boolean type, this will be serialized as string and needs to be converted back (`False` will be serialized as "False" which is a _truthy_ value).

One option here is to actually use numbers:
```python
In [1]: int(True)                                                                                                                                                                                               
Out[1]: 1

In [2]: int(False)                                                                                                                                                                                              
Out[2]: 0

In [3]: bool(int(True))                                                                                                                                                                                         
Out[3]: True

In [4]: bool(int(False))                                                                                                                                                                                        
Out[4]: False
```